### PR TITLE
(Step 5_3) Styled comment input and comment display and added comment deletion option.

### PR DESCRIPTION
**Pull request by cheoalee**
Used gson pretty print to format comment display, and allowed user to select number of comments to see. Also permitted user to clear all comments thus far in the Datastore.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -15,18 +15,160 @@
...
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comments, feeding into and reading from Datastore.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  // Number of comments the visitor chooses to see.
+  int visitorChoice;
```
Comment (adamtcohen) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: this isn't associated with a particular user in anyway right? i.e. one user setting this affects other users which would be undesired behavior.

This selection could be persisted on client side. I think the walkthrough suggests an approach to this.

**Comment by vidhyal** [comment]
Please resolve conflicts.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -15,18 +15,160 @@
...
+  }
+
+  /**
+   * Converts an ArrayList of Comments into a JSON string using the Gson library. Note: First added
+   * the Gson library dependency to pom.xml.
+   * @param comments Comments from the server.
+   * @return Message as a JSON.
+   */
+  private String convertToJsonUsingGson(List<Comment> comments) {
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
```
[the remark this answers is not in the document]
Comment (cheoalee) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:171]: Addressed offline

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (cheoalee) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:37]: Addressed offline

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -76,8 +80,8 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(taskEntity);
 
-    // Redirect back to the HTML page on which the comment was entered.
-    response.sendRedirect("/data.html");
+    // Bring user to comments section
```
Comment (vidhyal) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:83]:
```suggestion
    // Bring user to comments section.
```

**Pull request merged**